Compile Time "mixed declarations and code" Warning in multi_physical_planner.c

In compilation time, we have a warning about multi_physical_planner:
```
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -pg -DLINUX_PROFILE -ggdb -O0 -fno-omit-frame-pointer -fpic -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers -Wno-clobbered -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wmissing-declarations -Wmissing-prototypes -I '/home/eren/workspace/citus/src/include' -I. -I./ -I/usr/local/pgsql/include/server -I/usr/local/pgsql/include/internal -D_GNU_SOURCE  -I/usr/local/pgsql/include  -c -o planner/multi_physical_planner.o planner/multi_physical_planner.c -MMD -MP -MF .deps/multi_physical_planner.Po

planner/multi_physical_planner.c: In function ‘BuildMapMergeJob’:
planner/multi_physical_planner.c:1741:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   char basePartitionMethod PG_USED_FOR_ASSERTS_ONLY =
```

The mentioned code part is this:

```
...
		if (hasUninitializedShardInterval)
		{
			ereport(ERROR, (errmsg("cannot range repartition shard with "
								   "missing min/max values")));
		}

		/* this join-type currently doesn't work for hash partitioned tables */
		char basePartitionMethod PG_USED_FOR_ASSERTS_ONLY =
			PartitionMethod(baseRelationId);
		Assert(basePartitionMethod != DISTRIBUTE_BY_HASH);

		mapMergeJob->partitionType = RANGE_PARTITION_TYPE;
...
```

--- a/src/backend/distributed/planner/multi_physical_planner.c
+++ b/src/backend/distributed/planner/multi_physical_planner.c
@@ -1729,6 +1729,7 @@ BuildMapMergeJob(Query *jobQuery, List *dependedJobList, Var *partitionKey,
 		bool hasUninitializedShardInterval = false;
 		uint32 shardCount = cache->shardIntervalArrayLength;
 		ShardInterval **sortedShardIntervalArray = cache->sortedShardIntervalArray;
+		char basePartitionMethod PG_USED_FOR_ASSERTS_ONLY = 0;
 
 		hasUninitializedShardInterval = cache->hasUninitializedShardInterval;
 		if (hasUninitializedShardInterval)
@@ -1737,9 +1738,9 @@ BuildMapMergeJob(Query *jobQuery, List *dependedJobList, Var *partitionKey,
 								   "missing min/max values")));
 		}
 
+		basePartitionMethod = PartitionMethod(baseRelationId);
+
 		/* this join-type currently doesn't work for hash partitioned tables */
-		char basePartitionMethod PG_USED_FOR_ASSERTS_ONLY =
-			PartitionMethod(baseRelationId);
 		Assert(basePartitionMethod != DISTRIBUTE_BY_HASH);
 
 		mapMergeJob->partitionType = RANGE_PARTITION_TYPE;

